src/main/java/com/limechain/network/protocol/blockannounce/BlockAnnounceEngine.java

```diff
@@ -35,42 +34,44 @@ public class BlockAnnounceEngine {
     }};
 
     public void receiveRequest(byte[] msg, PeerId peerId, Stream stream) {
-        boolean hasKey = peerHandshakes.containsKey(peerId);
+        boolean connectedToPeer = connectionManager.isConnected(peerId);
+        boolean isHandshake = msg.length == HANDSHAKE_LENGTH;
 
-        //Decode handshake from Polkadot
-        if (!hasKey && msg.length == HANDSHAKE_LENGTH) {
-            ScaleCodecReader reader = new ScaleCodecReader(msg);
-            BlockAnnounceHandshake handshake = reader.read(new BlockAnnounceHandshakeScaleReader());
-            peerHandshakes.put(peerId, handshake);
-            log.log(Level.INFO, "Received handshake from " + peerId + "\n" +
-                    handshake);
-
-            writeHandshakeToStream(stream, peerId);
+        if(!connectedToPeer && !isHandshake) {
```

Comment:
Should we move the connectedToPeer check above so we don't log it as no handshake?